File: python_programs/static_allocation.py
import time

NUM_ELEMENTS = 100000
total_time = 0

for _ in range(10):
    start = time.perf_counter()
    array = [i for i in range(NUM_ELEMENTS)]
    end = time.perf_counter()
    total_time += (end - start) * 1e9

with open("./results/python_static_allocation_results.txt", "w") as f:
    f.write(f"Average: {total_time / 10:.6f} ns\n")

File: java_programs/StaticAllocation.java
import java.io.FileWriter;
import java.io.IOException;

public class StaticAllocation {
    public static void main(String[] args) throws IOException {
        int NUM_ELEMENTS = 100000;
        long totalTime = 0;

        for (int test = 0; test < 10; test++) {
            long start = System.nanoTime();
            int[] array = new int[NUM_ELEMENTS];
            for (int i = 0; i < NUM_ELEMENTS; i++) {
                array[i] = i;
            }
            long end = System.nanoTime();
            totalTime += (end - start);
        }

        FileWriter writer = new FileWriter("./results/java_static_allocation_results.txt");
        writer.write(String.format("Average: %.6f ns\n", totalTime / 10.0));
        writer.close();
    }
}
